freqtrade/strategy/interface.py

```diff
@@ -118,7 +121,7 @@ class IStrategy(ABC, HyperStrategyMixin):
     # Class level variables (intentional) containing
     # the dataprovider (dp) (access to other candles, historic data, ...)
     # and wallets - access to the current balance.
-    dp: Optional[DataProvider] = None
+    dp: DataProvider
```

Comment:
I'd keep this as optional from a typehint perspective.

while we provide dataprovider within freqtrade, that's not guaranteed when running in a notebook (for example) - where it's not even part of the docs, so a check `if self.dp` will still be beneficial, and having `Optional` here will allow a good editor to point this out.